core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -647,58 +541,225 @@ private static Expression checkExpressionPadTruncate(
     }
   }
 
+  private Expression translateCastToDate(RelDataType sourceType,
+      Expression operand, ConstantExpression format,
+      Supplier<Expression> defaultExpression) {
+
+    switch (sourceType.getSqlTypeName()) {
+    case CHAR:
+    case VARCHAR:
+      // If format string is supplied, parse formatted string into date
+      return Expressions.isConstantNull(format)
+          ? Expressions.call(BuiltInMethod.STRING_TO_DATE.method, operand)
+          : Expressions.call(Expressions.new_(BuiltInMethod.PARSE_DATE.method.getDeclaringClass()),
```

Comment:
The declaring class is always SqlFunctions right? Or no? If it's always the same we can just pass the class directly instead of calling this getter? Same comment for above

Reply:
The declaring functions are from `SqlFunctions` but might belong to subclasses, e.g.
```
FORMAT_DATE => SqlFunctions.DateFormatFunction.class
PARSE_TIME => SqlFunctions.DateParseFunction.class
```
my assumption is we need to pass the inner class name as the `targetExpression`, I am not completely aware of the use of this [`targetExpression`](https://github.com/apache/calcite/blob/7ca6fc4bdd3e1274a62f00f4c8d5830162a74879/linq4j/src/main/java/org/apache/calcite/linq4j/tree/MethodCallExpression.java#L33) attribute in the  `MethodCallExpression` class, if anyone else can confirm?

Reply:
Yeah that makes sense. I think explicitly passing the class is more readable, it tells the user exactly where the method is defined. I also think it conforms better with the rest of Calcite. Up to you though

Reply:
A few other SQL functions are implemented by non-static methods (generally to allow caching of expensive state). Do whatever those functions do.

Reply:
Thanks for the pointers, I couldn't find an explicit usage of a non-static built-in method in `Expressions.call()`, but saw `getDeclaringClass()` used to build the target expression so using that pattern; I think it's easier to trace to the class of the method call even though its a slight hit on readability.